xpallets/assets-registrar/src/lib.rs

```diff
@@ -300,17 +332,30 @@ impl<T: Config> Module<T> {
     /// Actually register an asset.
     fn apply_register(id: AssetId, asset: AssetInfo) -> DispatchResult {
         let chain = asset.chain();
-        AssetIdsOf::mutate(chain, |ids| {
+        AssetIdsOf::<T>::mutate(chain, |ids| {
             if !ids.contains(&id) {
                 ids.push(id);
             }
         });
 
-        AssetInfoOf::insert(&id, asset);
-        AssetOnline::insert(&id, true);
+        AssetInfoOf::<T>::insert(&id, Some(asset));
+        AssetOnline::<T>::insert(&id, true);
 
         RegisteredAt::<T>::insert(&id, frame_system::Pallet::<T>::block_number());
 
         Ok(())
     }
 }
+
+#[cfg(feature = "std")]
+impl GenesisConfig {
+    /// Direct implementation of `GenesisBuild::assimilate_storage`.
+    ///
+    /// Kept in order not to break dependency.
+    pub fn assimilate_storage<T: Config>(
+        &self,
+        storage: &mut sp_runtime::Storage,
+    ) -> Result<(), String> {
+        <Self as GenesisBuild<T>>::assimilate_storage(self, storage)
+    }
+}
```

Comment:
According to the suggestions of the Rust compiler, you may be able to replace the re-implementation ``assimilate_storage`` by adding use ``crate::mock::sp_api_hidden_includes_construct_runtime::hidden_include::traits::GenesisBuild;`` in mock.rs.

Reply:
Better to use ``frame_support::traits::GenesisBuild;``